test/support/schema.rb

```diff
@@ -151,6 +163,5 @@
 Schema = GraphQL::Schema.define do
   query QueryType
   mutation MutationType
-  query_execution_strategy GraphQL::Batch::ExecutionStrategy
-  mutation_execution_strategy GraphQL::Batch::MutationExecutionStrategy
+  lazy_resolve(GraphQL::Batch::Promise, :sync)
```

Comment:
I'll make a promise.rb release with https://github.com/lgierth/promise.rb/pull/24 so `lazy_resolve(Promise, :sync)` can be used here and Promise.all can be used in the resolvers.  This does make sense without that dependancy bump on promise.rb though